File: js/task1/S1_VAR_07.js
function toNumberOrNull(x) {
  if (typeof x !== 'string') {
      return null;
  }

  const converted = +x.trim();

  return !Number.isNaN(converted) && x.trim() !== '' 
      ? converted 
      : null;
}

console.log(toNumberOrNull("12"));     
console.log(toNumberOrNull("12.5"));   
console.log(toNumberOrNull(" 12 "));   
console.log(toNumberOrNull("12x"));    
console.log(toNumberOrNull(""));       
console.log(toNumberOrNull("abc"));

File: python/task1/S1_VAR_07.py
def toNumberOrNull(x):
  if not isinstance(x, str):
      return None

  s = x.strip()
  try:
      return float(s) if s != "" else None
  except:
      return None

print(toNumberOrNull("12"))
print(toNumberOrNull("12.5"))
print(toNumberOrNull(" 12 "))
print(toNumberOrNull("12x"))
print(toNumberOrNull(""))
print(toNumberOrNull("abc"))
